src/quick-lint-js/fe/parse-statement.cpp

```diff
@@ -273,7 +273,12 @@ bool parser::parse_and_visit_statement(
       this->skip();
       this->check_body_after_label();
       goto parse_statement;
-
+      // "async export f()" is not valid. It should be "export async f()"
+    case token_type::kw_export: {
+      this->diag_reporter_->report(
+          diag_async_export_method{async_token.span()});
```

Comment:
Must fix: Use designated initializers (i.e. named arguments). Once you do, you'll realize that `async_token.span()` is wrong; it only covers the `async` keyword, not both `async` and `export`. This should have been tested; see my comment in the tests.
```suggestion
          diag_async_export_method{.async_export = async_token.span()});
```
I think this code should look something like the following:
```suggestion
          diag_async_export_method{
            .async_export = source_code_span(async_token.begin(), this->peek().end),
          });
```